Polyhedron/demo/Polyhedron/MainWindow.cpp

```diff
@@ -1877,39 +1889,54 @@ void MainWindow::on_actionLoad_triggered()
   std::size_t nb_item = -1;
   
   Q_FOREACH(const QString& filename, dialog.selectedFiles()) {
-    
+
     CGAL::Three::Scene_item* item = NULL;
     if(selectedPlugin) {
       QFileInfo info(filename);
-      item = loadItem(info, selectedPlugin);
-      item->setColor(colors_[++nb_item]);
-      Scene::Item_id index = scene->addItem(item);
-      selectSceneItem(index);
-      CGAL::Three::Scene_group_item* group =
-          qobject_cast<CGAL::Three::Scene_group_item*>(item);
-      if(group)
-        scene->redraw_model();
+      bool ok;
+      QList<Scene_item*> result = loadItem(info, selectedPlugin, ok);
+      if(!ok)
+        continue;
+      for(Scene_item* item : result)
+      {
+        if(!item->property("already_colored").toBool())
+        {
+          ++nb_item;
+          item->setColor(colors_[nb_item]);
+        }
+        selectSceneItem(scene->item_id(item));
+        CGAL::Three::Scene_group_item* group =
+            qobject_cast<CGAL::Three::Scene_group_item*>(item);
+        if(group)
+          scene->redraw_model();
+      }
       this->addToRecentFiles(filename);
     } else {
       int scene_size = scene->numberOfEntries();
       open(filename);
-      if(scene->numberOfEntries() != scene_size)
-        scene->item(scene->numberOfEntries()-1)->setColor(colors_[++nb_item]);
+      item = scene->item(scene->numberOfEntries()-1);
+      if(scene->numberOfEntries() != scene_size
+         && !item->property("already_colored").toBool())
+        item->setColor(colors_[++nb_item]);
     }
   }
 }
 
 void MainWindow::on_actionSaveAs_triggered()
 {
-  Scene_item* item = NULL;
-
-  Q_FOREACH(Scene::Item_id id, scene->selectionIndices())
+  QList<Scene_item*> to_save;
+  for(Scene::Item_id id : scene->selectionIndices())
   {
-    item = scene->item(id);
+    Scene_item* item = scene->item(id);
+    to_save.append(item);
+  }
+  while(!to_save.empty())
+  {
+    Scene_item* item = to_save.front();
```

Comment:
I don't understand what you have in mind.

Reply:
`can_save` seems to be called on only one item of the list. I suggest that if selection size is 1 this is fine otherwise maybe a different function might to be called for a range (like `can_save_multiple_items()` for example).

Reply:
can_save is called for every item in the list `to_save`. A plugin that saved an item calls `pop_front()` on its input list, so I did not have to make different functions for saving one or several items.